control-plane/agents/src/bin/core/controller/wrapper.rs

```diff
@@ -833,13 +834,18 @@ impl ClientOps for Arc<tokio::sync::RwLock<NodeWrapper>> {
         self.update_replica_states(ctx.deref_mut()).await?;
         Ok(pool)
     }
-    /// Destroy a pool on the node via gRPC
+    /// Destroy a pool on the node via gRPC.
     async fn destroy_pool(&self, request: &DestroyPool) -> Result<(), SvcError> {
         let dataplane = self.grpc_client_locked(request.id()).await?;
-        let _ = dataplane.destroy_pool(request).await?;
+        let destroy_response = dataplane.destroy_pool(request).await;
         let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
         self.update_pool_states(ctx.deref_mut()).await?;
-        Ok(())
+        match destroy_response {
+            Ok(_) => Ok(()),
+            Err(_) => Err(SvcError::PoolNotFound {
+                pool_id: request.id.clone(),
+            }),
+        }
```

Comment:
I'm confused by this, seems all errors become a PoolNotFound error?

Reply:
Yes, At this time dataplane returns error in only 2 scenarios. 1. When uuid provided in DestroyPoolRequest doesnt match with uuid of the object in dataplane. 2. When Pool is not found
1st scenario is unlikely as we do not create Pools with uuid. DestroyPoolRequest is also sent with None for uuid.
However, It doesnt seem right to map any error to one type. So i will add logic in Control plane to check if pool exist and return error.

Reply:
I don't think we should make such assumptions, eg: the error may be a transport error or a timeout and in such case it's really confusing if we report a pool not found error.
I think we should simply return the error we find, and specifically map NotFound to Ok(()), for now. Later I think we probably want to handle notFound differently but for now we should map to Ok(()) to be compatible with the current code.

Reply:
Will look into modifying errors returned by dataplane.

Reply:
This PR https://github.com/openebs/mayastor/pull/1192/files will make sure dataplane returns more specific errors. Incorporated changes you requested here also.